dhall-docs/src/Dhall/Docs.hs

```diff
@@ -171,39 +199,54 @@ saveHtml inputAbsDir outputAbsDir t@(absFile, _) = do
 {-| Create an index.html file on each folder available in the second argument
     that lists all the contents on that folder.
 
-    Examples:
+    For example,
 
-    >>> createIndexes $(mkAbsDir "/") [
-        , $(mkAbsFile "/a/b.txt")
-        , $(mkAbsFile "/a/c/b.txt")
-        , $(mkAbsFile "/a/c.txt")
+    @
+    createIndexes [absdir|/|]
+        [ [absfile|/a/b.t`xt|]
+        , [absfile|/a/c/b.txt|]
+        , [absfile|/a/c.txt"|]
         ]
+    @
 
     ... will create two index.html files:
 
-    1. $(mkAbsFile "/a/index.html"), that will list the @"/a/b.txt"@ and
-    @"/a/c.txt"@ files
-    2. $(mkAbsFile "/a/c/index.html") that will list the @"a/c/b.txt"@ file
+    1. @[absfile|/a/index.html|]@, that will list the @/a/b.txt@ and
+    @/a/c.txt@ files
+    2. @[absfile|/a/c/index.html|]@ that will list the @a/c/b.txt@ file
 
 -}
 createIndexes
     :: Path Abs Dir    -- ^ Directory where index.html file will be saved. Used
                        --   to link the css resources. It should be a prefix for
                        --   each @Path Abs File@ on the second argument
     -> [Path Abs File] -- ^ Html files generated by the tool
+    -> String          -- ^ Package name
     -> IO ()
-createIndexes outputPath htmlFiles = do
+createIndexes outputPath htmlFiles packageName = do
     let toMap file = Map.singleton (Path.parent file) [file]
     let filesGroupedByDir = Map.unionsWith (<>) $ map toMap htmlFiles
 
+    let listDirRel dir = do
+            dirs <- fst <$> Path.IO.listDir dir
+            mapM (Path.stripProperPrefix dir) dirs
+
     let createIndex index files = do
             indexFile <- Path.fromAbsFile . (index </>) <$> Path.parseRelFile "index.html"
-            let relativeResources = resolveRelativePath outputPath index
+            indexTitle <-
+                if outputPath == index then return "package"
+                else Path.fromRelDir <$> Path.stripProperPrefix outputPath index
+            indexList <- Control.Monad.forM files $
+                fmap Path.filename . Path.stripProperPrefix outputPath
+            dirList <- listDirRel index
+
+            let relativeResourcesPath = resolveRelativePath outputPath index
             Lucid.renderToFile indexFile $
                 indexToHtml
-                    (Path.fromAbsDir index)
-                    (map Path.fromAbsFile files)
-                    (relativeResources <> "index.css")
+                    indexTitle
+                    indexList
+                    dirList
+                    DocParams {..}
```

Comment:
Personally, I think it's hard to see what that `DocParams` record will contain, particularly when there's so much else going on in the surrounding code.

Reply:
Yes you're right. I updated it so that arguments are explicitly provided